### DISCUSSION TITLE
Columns added in code behind

### DISCUSSION THREAD

**Discussion by chemajimenezarup** [macgile/DataGridFilter#169] [General]
Hello,
I am adding columns dynamically from the code behind and, for those columns, I don't get the filter button. Is there any way to do it?
Thanks in advance!!

**Comment by macgile** [comment]
Hi,

You can try this 

            Binding bName = new Binding("Name");

            var textColumn = new FilterDataGrid.DataGridTextColumn() { Header = "Name", IsColumnFiltered = true, Binding = bName };
            MyFilterDataGrid.AutoGenerateColumns = false;
            MyFilterDataGrid.Columns.Add(textColumn);


best regards
